src/groups/FBX.tsx

```diff
@@ -0,0 +1,245 @@
+      }
+
+      // Material Props
+      if (materialProps) {
+        const { color, ...otherMaterialProps } = materialProps;
+        updateAllMaterials(color, fbx.scene.children, otherMaterialProps);
+      }
+
+      // Animation
+      /* eslint-disable-next-line no-param-reassign */
+      mixer.current = new THREE.AnimationMixer(fbx);
+      if (fbx.animations && fbx.animations.length > 0) {
+        const action = mixer.current.clipAction(fbx.animations[actionIndex]);
+        action.play();
+      }
+
+      // Env Map + Shadows
+      fbx.traverse(mesh => {
+        /* eslint-disable no-param-reassign */
+        // @ts-ignore
+        if (mesh.isMesh) {
+          if (envMap) {
+            // @ts-ignore
+            mesh.material.envMap = envMap;
+          }
+          mesh.castShadow = castShadow;
+          mesh.receiveShadow = receiveShadow;
+        }
+        /* eslint-enable no-param-reassign */
+      });
+
+      // Add FBX
+      group.current.add(fbx);
+    }
+  });
+}
+
+const FBX: React.FunctionComponent<FBXProps> = function FBX({
+  fbxPath = "",
+  fbxURL,
+  view3DEnvMap = false,
+  castShadow = false,
+  receiveShadow = false,
+  ...otherProps
+}) {
+  const group = useRef();
+  const mixer = useRef<THREE.AnimationMixer>();
+  const { envMap } = useViewContext();
+
+  // Material Props
+  const materialProps = useMemo(
+    function updateMaterialProps() {
+      // No Material Props
+      if (!otherProps || Object.keys(otherProps).length === 0) {
+        return undefined;
+      }
+
+      // Acceptable Material Props
+      const materialPropsKeys = [
+        "color",
+        "wireframe",
+        "opacity",
+        "transparent",
+        "side",
+        "depthWrite",
+        "depthTest"
+      ];
+
+      // Extract Acceptable Props
+      const props = {};
+      materialPropsKeys.map(propName => {
+        if (otherProps[propName]) {
+          props[propName] = otherProps[propName];
+        }
+
+        return null;
+      });
+
+      return Object.keys(props).length > 0 ? props : undefined;
+    },
+    [otherProps]
+  );
+
+  // Environment Map
+  const _envMap = useMemo(
+    function updateEnvMap() {
+      if (view3DEnvMap) {
+        return envMap;
+      }
+
+      return null;
+    },
+    [view3DEnvMap, envMap]
+  );
+
+  // Loading Text
+  const loadingText = useMemo(
+    function updateLoadText() {
+      return fbxURL || "No fbxURL";
+    },
+    [fbxURL]
+  );
+
+  // Load FBX and Animation Mixer
+  useEffect(
+    function updateFBXandMixer() {
+      if (group) {
+        loadFBX({
+          group,
+          fbxPath,
+          fbxURL,
+          mixer,
+          envMap: _envMap,
+          castShadow,
+          receiveShadow,
+          materialProps
+        });
+      }
+    },
+    [group, fbxPath, fbxURL, _envMap, castShadow, receiveShadow, materialProps]
+  );
+
+  // Animation Mixer Update
+  const clock = new THREE.Clock();
+  useAnimationFrame(function updateMixer() {
```

Comment:
does it make sense to use `useFrame()` here

Reply:
Yeah, this needs to execute every frame for the animations to work. Otherwise the FBX models will just be still.

Reply:
yea `useFrame` executes just before every animation frame.

Reply:
Ah, sorry, I misread `useFrame` for `useAnimationFrame`. Yeah, `useFrame` works here.